api/valor_api/backend/metrics/llm_call.py

```diff
@@ -0,0 +1,266 @@
+from typing import Any
+
+import openai
+from mistralai.client import MistralAPIException, MistralClient
+from mistralai.models.chat_completion import ChatMessage
+
+COHERENCE_INSTRUCTION = """You are a helpful assistant. You will grade the user's text. Your task is to rate the text based on its coherence. Please make sure you read and understand these instructions carefully. Please keep this document open while reviewing, and refer to it as needed.
+Evaluation Criteria:
+Coherence (1-5) - the collective quality of all sentences. We align this dimension with the DUC quality question of structure and coherence whereby ”the summary should be well-structured and well-organized. The summary should not just be a heap of related information, but should build from sentence to sentence to a coherent body of information about a topic.”
+Evaluation Steps:
+1. Read the text carefully and identify the main topic and key points.
+2. Assign a score for coherence on a scale of 1 to 5, where 1 is the lowest and 5 is the highest based on the Evaluation Criteria. Respond with just the number 1 to 5."""
+
+
+class LLMClient:
+    """
+    Wrapper for calls to an LLM API.
+
+    Parameters
+    ----------
+    api_key : str, optional
+        The API key to use.
+    model_name : str
+        The model to use.
+    """
+
+    api_key: str | None = None
+    model_name: str
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        model_name: str | None = None,
+    ):
+        """
+        Set the API key and model name (if provided).
+        """
+        self.api_key = api_key
+        if model_name is not None:
+            self.model_name = model_name
+
+    def connect(
+        self,
+    ):
+        """
+        Setup the connection to the API.
+        """
+        raise NotImplementedError
+
+    def process_messages(
+        self,
+        messages: list[dict[str, str]],
```

Comment:
Should we use stricter validation here? messages should be in the OpenAI format, which means that the keys for the dictionaries should be "role" and "content". What would be the best way to validate this?